### PR TITLE
评估器

### LINKED ISSUE #71: Estimators
* Markdown文件：[文件地址](https://github.com/xitu/tensorflow/blob/zh-hans/tensorflow/docs_src/programmers_guide/estimators.md)
  * PR 地址：#99
  ----
  * 翻译时间：2 天
  * 校对时间：1 天
  * 翻译积分：4 分
  * 校对积分：1.5 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by edvardHua**
完成翻译@leviding，resolve #71

### CODE REVIEW

**Comment by MRNIU** [comment]
@leviding 校对认领

**Comment by leviding** [comment]
@MRNIU ok

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 1]
```diff
@@ -1,176 +1,125 @@
...
-own custom Estimators.  All Estimators--whether pre-made or custom--are
-classes based on the @{tf.estimator.Estimator} class.
+# 评估器
+
+这篇文档介绍的 @{tf.estimator$**Estimator**} 是一个高层的 TensorFlow API，它极大的简化了机器学习编程工作。评估器封装的行为如下：
+
+* 训练
+* 评估
+* 预测
+* 导出作为服务
```
Comment (MRNIU) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 导出作为服务 -> 作为服务导出

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 2]
```diff
@@ -1,176 +1,125 @@
...
-    build the graph.
-*   Estimators provide a safe distributed training loop that controls how and
-    when to:
-    *   build the graph
-    *   initialize variables
-    *   start queues
-    *   handle exceptions
-    *   create checkpoint files and recover from failures
-    *   save summaries for TensorBoard
+* 基于评估器的模型可以运行在单机上，也可以不做修改的运行在分布式多服务器环境。甚至，你还可以将基于评估器的模型运行 CPUs，GPUs 或者 TPUs 上。
```
Comment (MRNIU) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 你还可以将基于评估器的模型运行 -> 你还可以将基于评估器的模型运行在

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 3]
```diff
@@ -1,176 +1,125 @@
...
+* 评估器能让模型开发人员共享内部实现。
```
Comment (MRNIU) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 评估器能让模型开发人员共享内部实现。 -> 评估器简化了模型开发人员内部共享的实现。

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 4]
```diff
@@ -1,176 +1,125 @@
...
 
-When writing an application with Estimators, you must separate the data input
-pipeline from the model.  This separation simplifies experiments with
-different data sets.
+当你用评估器写应用时，你必须将数据输入管道和模型分开。这种分离简化了不同数据集的实验。
 
+## 预定义的评估器
 
-## Pre-made Estimators
+比起 TensorFlow 的基本 API，预定义的评估器可以让你在更高抽象的层面上工作。你不再需要操心的创建计算图和会话，因为评估器已经帮你把这一切都`串通`好了。也就是说，预定义的评估器已经帮你创建和管理@{tf.Graph$`图`}和@{tf.Session$`会话`}对象。甚至，预定义的评估器可以让你修改最少的代码来试验不同的模型架构。譬如@{tf.estimator.DNNClassifier$`DNNClassifier`}就是一个预定义的评估器，它可以训练密集的前向传递神经网络分类模型。
```
Comment (MRNIU) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 你不再需要操心的创建计算图和会话 -> 你不再需要操心创建计算图和会话
 "plumbing"  不需要翻译
这一段里面的空格请加一下

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 5]
```diff
@@ -1,176 +1,125 @@
...
+1. **编写一个或多个数据集的导入函数。**举个例子，你可能会创建一个函数用于导入训练数据和另外一个函数用于导入测试数据。每一个数据集的导入函数都会返回下面两个对象：
+    
+    * 一个字典，它的 key 是特征名，而 values 是对应的张量（或者是稀疏张量），张量里面包含了对应的特征数据。
+    * 一个张量，它包含了一个或多个标签。
+    
+    举个例子，下面的代码是一个输入函数的基本框架：
+    
+    ```python
+    def input_fn(dataset):
+    	... # manipulate dataset, extracting feature names and the label
```
Comment (MRNIU) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 注释需要翻译

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 6]
```diff
@@ -1,176 +1,125 @@
...
+    	return feature_dict, label
+    ```
+    
+    更多的细节，请看@{$programmers_guide/datasets}。
```
Comment (MRNIU) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 请看@{$programmers_guide/datasets}。 -> 请看 @{$programmers_guide/datasets} 。

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 7]
```diff
@@ -1,176 +1,125 @@
...
-# Treat the derived Estimator as you would any other Estimator. For example,
-# the following derived Estimator calls the train method:
+# 像使用预定义评估器那样使用这个派生评估器。举个例子,
+# 我们可以调用派生评估器的训练方法。
 est_inception_v3.train(input_fn=my_training_set, steps=2000)
 ```
-For more details, please refer to the documentation for
-@{tf.keras.estimator.model_to_estimator}.
+
+想要了解更多的细节，请查阅@{tf.keras.estimator.model_to_estimator}。
```
Comment (MRNIU) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 请查阅@{tf.keras.estimator.model_to_estimator}。 -> 请查阅 @{tf.keras.estimator.model_to_estimator} 。

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 8]
```diff
@@ -1,176 +1,125 @@
...
-@{tf.keras.estimator.model_to_estimator} as in the
-following sample:
+你可以将 Keras 模型转换成评估器。这样 Keras 模型就可以利用到评估器的优点了，譬如分布式训练。调用 @{tf.keras.estimator.model_to_estimator} 就可达到目的，代码如下：
 
 ```python
-# Instantiate a Keras inception v3 model.
+# 实例化一个 kera inception v3 模型。
 keras_inception_v3 = tf.keras.applications.inception_v3.InceptionV3(weights=None)
 # Compile model with the optimizer, loss, and metrics you'd like to train with.
+# 定义好用来训练模型使用的优化器，损失和评价指标，然后再编译它
```
Comment (MRNIU) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 原文注释可以删掉了

**Comment by MRNIU** [comment]
@edvardHua @leviding 校对完成

**Comment by Romeo0906** [comment]
认领校对 @leviding

**Comment by leviding** [comment]
@Romeo0906 ok

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 9]
```diff
@@ -1,176 +1,125 @@
...
-*   evaluation
-*   prediction
-*   export for serving
-
-You may either use the pre-made Estimators we provide or write your
-own custom Estimators.  All Estimators--whether pre-made or custom--are
-classes based on the @{tf.estimator.Estimator} class.
+# 评估器
+
+这篇文档介绍的 @{tf.estimator$**Estimator**} 是一个高层的 TensorFlow API，它极大的简化了机器学习编程工作。评估器封装的行为如下：
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 高层的 API => 高级的 API
 评估器封装的行为如下 => 评估器封装了如下行为

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 10]
```diff
@@ -1,176 +1,125 @@
...
+
+* 训练
+* 评估
+* 预测
+* 导出作为服务
 
-Note: TensorFlow also includes a deprecated `Estimator` class at
-@{tf.contrib.learn.Estimator}, which you should not use.
+你可以使用我们预定义好的评估器，也可以自己写一个评估器。但预定义和自定义的评估器都是基于 @{tf.estimator.Estimator} 类的。
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 但无论预定义的还是自定义的评估器都继承自 @{tf.estimator.Estimator} 类。

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 11]
```diff
@@ -1,176 +1,125 @@
...
 
+注意：TensorFlow 还包含了一个已经被抛弃的 `Estimator` 类 @{tf.contrib.learn.Estimator}，我们不要去使用它。
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 被抛弃的 => 被弃用的

File: tensorflow/docs_src/programmers_guide/estimators.md (hunk 2 above)
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 也可以不做修改的运行在分布式多服务器环境。=>  也可以运行在分布式的多台服务器环境上并且不需要做任何修改。
甚至，你还可以将基于评估器的模型运行 CPUs，GPUs 或者 TPUs 上。 => 更棒的是，你还可以将其运行在 CPU, GPU 或者 TPU 上并且不用做任何标记。

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 12]
```diff
@@ -1,176 +1,125 @@
...
-    when to:
-    *   build the graph
-    *   initialize variables
-    *   start queues
-    *   handle exceptions
-    *   create checkpoint files and recover from failures
-    *   save summaries for TensorBoard
+* 基于评估器的模型可以运行在单机上，也可以不做修改的运行在分布式多服务器环境。甚至，你还可以将基于评估器的模型运行 CPUs，GPUs 或者 TPUs 上。
+* 评估器能让模型开发人员共享内部实现。
+* 你可以用直观的高层代码编写先进模型。简而言之，使用评估器创建模型比使用 TensorFlow 的低层 API 更加简单。
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 你可以用高级的直观的代码编写某种状态的艺术模型。简而言之，使用评估器通常会比使用 TensorFlow 的低级 API 更加简捷。

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 13]
```diff
@@ -1,176 +1,125 @@
...
+* 评估器是建立在 tf.layers 上的，所以它让定制更加便捷。
+* 评估器已经为你构建了图。换句话说，你不需要构建图了。
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 图 => 图表

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 14]
```diff
@@ -1,176 +1,125 @@
...
-    *   build the graph
-    *   initialize variables
-    *   start queues
-    *   handle exceptions
-    *   create checkpoint files and recover from failures
-    *   save summaries for TensorBoard
+* 基于评估器的模型可以运行在单机上，也可以不做修改的运行在分布式多服务器环境。甚至，你还可以将基于评估器的模型运行 CPUs，GPUs 或者 TPUs 上。
+* 评估器能让模型开发人员共享内部实现。
+* 你可以用直观的高层代码编写先进模型。简而言之，使用评估器创建模型比使用 TensorFlow 的低层 API 更加简单。
+* 评估器是建立在 tf.layers 上的，所以它让定制更加便捷。
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 所以它让定制更加便捷。 => 简化了自定义的内容。

Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 所以它让定制更加便捷。 => 简化了自定义的内容。

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 15]
```diff
@@ -1,176 +1,125 @@
...
+* 评估器已经为你构建了图。换句话说，你不需要构建图了。
+* 评估器提供了一个安全的分布式训练的循环，它能够控制怎么样以及何时运行：
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 它能够控制怎么样以及何时运行 => 能够控制以下操作的运行时间和运行方式

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 16]
```diff
@@ -1,176 +1,125 @@
...
+  * 创建图
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 构建图表

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 17]
```diff
@@ -1,176 +1,125 @@
...
+  * 初始化变量
+  * 启动队列
+  * 处理异常
+  * 创建检查点文件和从错误中恢复
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 创建校验文件和错误恢复

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 18]
```diff
@@ -1,176 +1,125 @@
...
-# Instantiate a Keras inception v3 model.
+# 实例化一个 kera inception v3 模型。
 keras_inception_v3 = tf.keras.applications.inception_v3.InceptionV3(weights=None)
 # Compile model with the optimizer, loss, and metrics you'd like to train with.
+# 定义好用来训练模型使用的优化器，损失和评价指标，然后再编译它
 keras_inception_v3.compile(optimizer=tf.keras.optimizers.SGD(lr=0.0001, momentum=0.9),
                           loss='categorical_crossentropy',
                           metric='accuracy')
-# Create an Estimator from the compiled Keras model.
+# 从编译好的模型转换成评估器
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 从编译好的模型转换成评估器 => 从编译好的 Keras 模型创建一个评估器

File: tensorflow/docs_src/programmers_guide/estimators.md [hunk 19]
```diff
@@ -1,176 +1,125 @@
...
 est_inception_v3 = tf.keras.estimator.model_to_estimator(keras_model=keras_inception_v3)
-# Treat the derived Estimator as you would any other Estimator. For example,
-# the following derived Estimator calls the train method:
+# 像使用预定义评估器那样使用这个派生评估器。举个例子,
+# 我们可以调用派生评估器的训练方法。
```
Comment (Romeo0906) [inline tensorflow/docs_src/programmers_guide/estimators.md]: 像使用预定义评估器那样使用这个派生评估器。举个例子,
我们可以调用派生评估器的训练方法。
=>
像使用其他评估器一样使用派生的评估器。以下例子中，派生的评估器调用了训练方法：

**Comment by Romeo0906** [comment]
@leviding  @edvardHua  校对完成，译者翻译的很赞！

**Comment by leviding** [comment]
赞 😘， @edvardHua 可以修改啦

各位辛苦啦，有空的话多翻译校对几个，咱们马上搞完了 😎

**Comment by edvardHua** [comment]
@leviding 改完啦。

**Comment by leviding** [comment]
辛苦啦
